controller/array_action/array_mediator_abstract.py

```diff
@@ -21,32 +22,34 @@ def __init__(self, user, password, endpoint):  # pylint: disable=super-init-not-
 
     @retry(NoConnectionAvailableException, tries=11, delay=1)
     def map_volume_by_initiators(self, vol_id, initiators):
-        host_name, connectivity_types = self.get_host_by_host_identifiers(initiators)
-
-        logger.debug("hostname : {}, connectivity_types  : {}".format(host_name, connectivity_types))
-
-        connectivity_type = utils.choose_connectivity_type(connectivity_types)
-        if NVME_OVER_FC_CONNECTIVITY_TYPE == connectivity_type:
-            array_initiators = []
-        elif FC_CONNECTIVITY_TYPE == connectivity_type:
-            array_initiators = self.get_array_fc_wwns(host_name)
-        elif ISCSI_CONNECTIVITY_TYPE == connectivity_type:
-            array_initiators = self.get_iscsi_targets_by_iqn(host_name)
-        else:
-            raise UnsupportedConnectivityTypeError(connectivity_type)
-
         mappings = self.get_volume_mappings(vol_id)
         if len(mappings) >= 1:
             logger.debug(
                 "{0} mappings have been found for volume. the mappings are: {1}".format(len(mappings), mappings))
             if len(mappings) == 1:
-                mapping = list(mappings)[0]
-                if mapping == host_name:
+                mapping_host_name = list(mappings)[0]
+                host = self.get_host_by_name(mapping_host_name)
+                if host.initiators in initiators:
                     logger.debug("idempotent case - volume is already mapped to host.")
-                    return mappings[mapping], connectivity_type, array_initiators
-            raise array_errors.VolumeMappedToMultipleHostsError(mappings)
+                    lun = mappings[mapping_host_name]
+                    logger.debug(
+                        "hostname : {}, connectivity_types  : {}".format(host.host_name, host.connectivity_types))
+                    connectivity_type = utils.choose_connectivity_type(host.connectivity_types)
+                    array_initiators = self._get_array_initiators(host.host_name, connectivity_type)
```

Comment:
I know it's existing naming, but:
```suggestion
                    array_targets = self._get_array_targets(host.host_name, connectivity_type)
```
AFAIK, the array side has the targets, not the initiators

Reply:
targets is too general IMO

Reply:
it's more accurate than "initiators", that's for sure.
I thought "targets" are too specific, because I think it's an iSCSI term (same for "initiators").
do you have a better suggestion?

Reply:
CSI-4142